### PR TITLE
cancel-redundant-builds

### PR DESCRIPTION
**Pull request by axrs**

### CODE REVIEW

File: cancel-redundant-builds.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+if [[ -n "$CIRCLECI" ]];then
+	BASE_API_URL="https://circleci.com/api/v1.1/project/github/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/tree/$CIRCLE_BRANCH"
```
Comment (AndreTheHunter) [inline cancel-redundant-builds.sh]: Put the VCS type as a variable too https://circleci.com/docs/api/v1-reference/#version-control-system-vcs-type
CircleCI doesn't provide one so just make your own env var. e.g. `export PROJECT_VCS_TYPE='github'`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,32 @@
...
+
+### Example CircleCI config
+
+```
+jobs:
+  test:
+    docker:
+      - image: jesiio/docker-git-awscli:latest
+    steps:
+      - run: 'wget -O - https://raw.githubusercontent.com/jesims/circleci-tools/master/cancel-redundant-builds.sh | bash'
```
Comment (AndreTheHunter) [inline README.md]: Could use `curl` here, may be simpler.
Wget is really better when you just want to easily download the file, and curl would be on more CircleCI images

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    docker:
+      - image: jesiio/docker-git-awscli:latest
+    steps:
+      - checkout
+      - run: './cancel-redundant-builds.sh'
+      - run: 'sleep 1m'
+      - run: './cancel-redundant-builds.sh'
+      - run: 'sleep 1m'
+      - run: './cancel-redundant-builds.sh'
+      - run: 'sleep 1m'
```
Comment (AndreTheHunter) [inline .circleci/config.yml:14]: TIL BusyBox `sleep` can take an optional suffix, unix `sleep` does not

File: README.md (hunk 2 above)
Comment (AndreTheHunter) [inline README.md]: Could use `curl` here, may be simpler.
Wget is really better when you just want to easily download the file, and curl would be on more CircleCI images

In our case, I think we should just put the script in our CI images

Comment (AndreTheHunter) [inline README.md]: Could use `curl` here, may be simpler.
Wget is really better when you just want to easily download the file, and curl would be on more CircleCI images

File: test.json [hunk 4]
```diff
@@ -0,0 +1,1178 @@
...
+    "build_time_millis": 9609,
+    "start_time": "2018-02-06T01:47:23.996Z",
+    "platform": "2.0",
+    "outcome": "success",
+    "vcs_url": "https://github.com/jesims/jesi",
+    "author_name": "Alexander Scott",
+    "queued_at": "2018-02-06T01:47:22.255Z",
+    "author_email": "xander@cadcoder.com"
+  }
+]
```
Comment (AndreTheHunter) [inline test.json]: There's no private information in here right?

**Review by AndreTheHunter — CHANGES_REQUESTED** [review]

File: test.json (hunk 4 above)
Comment (axrs) [reply test.json]: Nothing really scary. A private github link to pull requests and commit hashes. Some random commit messages with JIRA numbers. etc.

I'll clean and sanitise

**Review by AndreTheHunter — APPROVED** [review]

**Pull request merged**